include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -64,6 +67,28 @@ namespace dht
 		void reset();
 	};
 
+	struct dht_immutable_data
+	{
+		sha1_hash target;
+		std::vector<char> value;
+		std::array<char, 128> ips;
+		time_point last_seen;
+	};
+
+	struct dht_mutable_data : dht_immutable_data
+	{
+		std::array<char, signature::len> signature;
```

Comment:
yes, well they are full types (in `types.hpp`), but not sure if you want to put this in the `dht_state`, because then, the question will be about the `bloom_filter`.